### PR TITLE
LPD-24964 System/instance configurations that are hidden from the interface can be accessible by its URL

### PR DESCRIPTION
**Pull request by thiago-buarqque**
https://liferay.atlassian.net/browse/LPD-24964

# Issue
It is possible to access a System/Instance configuration when it's supposed to be hidden from the UI.

## Steps to reproduce
**Preconditions**

Make sure the Script Management configuration IS NOT visible on the System Configurations page. If it’s visible, find a system configuration that is not accessible via interface and change script-management on step 1 URL to configuration-key-you-found.

**Steps**

1. Access the [script management configuration URL](http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_configuration_admin_web_portlet_SystemSettingsPortlet&p_p_lifecycle=0&p_p_state=maximized&p_p_mode=view&_com_liferay_configuration_admin_web_portlet_SystemSettingsPortlet_mvcRenderCommandName=%2Fconfiguration_admin%2Fview_configuration_screen&_com_liferay_configuration_admin_web_portlet_SystemSettingsPortlet_configurationScreenKey=script-management) (or other configuration in case someone make this one visible in the future).

# Proposed fix
Before render the UI, check if the configuration should be visible, if not, prevent rendering.

## Preview of the proposed fix
image

### CODE REVIEW

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:sf

File: modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/portlet/action/ViewConfigurationScreenMVCRenderCommand.java [hunk 1]
```diff
@@ -57,6 +58,14 @@ public String render(
 			_configurationEntryRetriever.getConfigurationScreen(
 				configurationScreenKey);
 
+		if (!configurationScreen.isVisible()) {
+			throw new PortletException(
+				StringBundler.concat(
+					"The ", configurationScreen.getScope(), " configuration \"",
+					configurationScreen.getName(themeDisplay.getLocale()),
+					"\" is not accessible through the UI"));
```
Comment (ethib137) [inline modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/portlet/action/ViewConfigurationScreenMVCRenderCommand.java:66]: Hey @thiago-buarqque , this is looking good, but I'm unsure about the phrase "through the UI"? Why not just have "is not accessible" or "is not visible"? What do you think?

Comment (thiago-buarqque) [reply modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/portlet/action/ViewConfigurationScreenMVCRenderCommand.java:66]: @ethib137 that makes sense. I'll go with "is not accessible"

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Review by ethib137 — APPROVED** [review]
LGTM... If there aren't any related test failures, feel free to forward.

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by ethib137** [comment]
ci:test:relevant

**Pull request closed without merging**